### PR TITLE
setup.py: Add installation script.

### PR DESCRIPTION
**Pull request by grumbling-tom**
Creates a `setup.py` file, as per #44 

This is also can also be `pip install`ed from TestPyPI using the steps [here](https://packaging.python.org/tutorials/packaging-projects/).

I've also created a Dockerfile to demonstrate the build's reproducibility, which can be seen [here](https://github.com/grumbling-tom/tiledb_netcdf/blob/feature/dockerfile/Dockerfile). Given the number of C library dependencies, it might be worth including in this PR, purely for illustration on installing those libraries.

### CODE REVIEW

**Comment by DPeterK** [comment]
Hi @grumbling-tom, thanks so much for putting this together! And sorry that I've only just found it - for some reason I was never notified that you'd put up this PR 😑 I'll try and take a look this week...

**Comment by grumbling-tom** [comment]
No problem! Let me know if I can help in any way.

File: setup.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+        "Programming Language :: Python :: 3",
+        "License :: OSI Approved :: GNU General Public License v3 or later (GPLv3+)",
+        "Operating System :: OS Independent",
+    ],
+    python_requires=">=3.7",
+    install_requires=[
+        "netCDF4>=1.5.3",
+        "tiledb>=0.6.2",
+        "numpy>=1.18.5",
+        "scitools-iris>=2.4.0",
```
Comment (DPeterK) [inline setup.py:27]: Iris _should_ itself provide the NetCDF4 and NumPy dependencies, so we might be able to drop at least NumPy from this list - maybe?

File: setup.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    packages=setuptools.find_packages(),
+    classifiers=[
+        "Programming Language :: Python :: 3",
+        "License :: OSI Approved :: GNU General Public License v3 or later (GPLv3+)",
+        "Operating System :: OS Independent",
+    ],
+    python_requires=">=3.7",
+    install_requires=[
+        "netCDF4>=1.5.3",
+        "tiledb>=0.6.2",
```
Comment (DPeterK) [inline setup.py:25]: The latest updates I've made mean we could probably bump this dependency now: 
```suggestion
        "tiledb>=0.6.6",
```

**Review by DPeterK — COMMENTED** [review]
Thanks again for providing this setup script, @grumbling-tom, it's a really useful addition. I've made just a couple of comments relating to the install requirements, but I think they're only minor. I've also confirmed the setup script works by installing into a conda env that I'd pre-populated with these dependencies 🙂 

I do accept though that Iris in particular is a pain to install without help, so I like the idea of including the Dockerfile you linked above. Perhaps you could add that in an `etc` folder to the library root, along with a quick readme explaining why it's provided? Longer-term I'd like to reduce the number of dependencies this library has, basically to only NetCDF and TileDB, and have extra functionality that's only enabled if the extra libraries (such as Xarray and Iris) are available to be imported. That's still in the future though, and definitely isn't a reason to further hold this PR up...

File: setup.py (hunk 1 above)
Comment (grumbling-tom) [reply setup.py:25]: Yep! Looks to have installed those ok in the resultant container:

```bash
$ docker build -f etc/Dockerfile -t tiledb_netcdf .
$ docker run --entrypoint bash tiledb_netcdf -c "pip freeze"
antlr4-python3-runtime==4.7.2
asciitree==0.3.3
Cartopy==0.18.0
certifi==2020.6.20
cf-units==2.1.4
cftime==1.2.1
cycler==0.10.0
dask==2.30.0
fasteners==0.15
kiwisolver==1.2.0
matplotlib==3.3.2
monotonic==1.5
netCDF4==1.5.4
numcodecs==0.7.2
numpy==1.18.5
pandas==1.1.3
Pillow==8.0.0
pyke==1.1.1
pyparsing==2.4.7
pyshp==2.1.2
python-dateutil==2.8.1
pytz==2020.1
PyYAML==5.3.1
scipy==1.5.3
scitools-iris==2.4.0
Shapely==1.7.1
six==1.15.0
tiledb==0.6.6
tiledb-netcdf @ file:///tiledb_netcdf
toolz==0.11.1
xarray==0.16.1
zarr==2.5.0
```

**Comment by grumbling-tom** [comment]
Thanks for the review! I've included that Dockerfile at the path `etc/Dockerfile` :whale:

**Review by DPeterK — APPROVED** [review]

**Comment by DPeterK** [comment]
Great stuff! Thanks @grumbling-tom 😄

**Pull request merged**